packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart

```diff
@@ -100,22 +108,359 @@ class VMInfoList extends StatelessWidget {
+  if (objectRef is LibraryRef) objectRefName = objectRef.name;
+
+  return objectRefName;
+}
+
+/// Recursively gets the full owner name of a field or function object.
+String? _ownerName(ObjRef? ref) {
+  if (ref == null) return '';
+  if (ref is FuncRef) {
+    return '${_ownerName(ref.owner)}.${_objectName(ref)}';
+  } else if (ref is FieldRef) {
+    return '${_ownerName(ref.owner)}.${_objectName(ref)}';
+  } else {
+    return _objectName(ref) ?? '<unknown>';
   }
+}
 
-  Widget _buildAlternatingRow(BuildContext context, int index, Widget row) {
-    return Container(
-      color: alternatingColorForIndex(index, Theme.of(context).colorScheme),
-      height: defaultRowHeight,
-      padding: const EdgeInsets.symmetric(
-        horizontal: defaultSpacing,
+/// An ExpansionTile with an AreaPaneHeader as header and custom style
+/// for the VM tools tab.
+class VmExpansionTile extends StatelessWidget {
+  const VmExpansionTile({
+    required this.title,
+    required this.children,
+    this.onExpanded,
+  });
+
+  final String title;
+  final List<Widget> children;
+  final void Function(bool)? onExpanded;
+
+  @override
+  Widget build(BuildContext context) {
+    final titleRow = AreaPaneHeader(
+      title: Text(title),
+      needsTopBorder: false,
+    );
+    return Theme(
+      data: Theme.of(context).copyWith(dividerColor: Colors.transparent),
+      child: ExpansionTile(
+        title: titleRow,
+        onExpansionChanged: onExpanded,
+        tilePadding: const EdgeInsets.all(4.0),
+        children: children,
       ),
-      child: row,
     );
   }
 }
+
+/// An expandable list to display the retaining objects for a given RetainingPath.
+class RetainingPathWidget extends StatelessWidget {
+  const RetainingPathWidget({
+    required this.fetching,
+    required this.retainingPath,
+    this.onExpanded,
+  });
+
+  final ValueListenable<bool> fetching;
+  final ValueListenable<RetainingPath?> retainingPath;
+  final void Function(bool)? onExpanded;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<bool>(
+      valueListenable: fetching,
+      builder: (context, fetching, child) {
+        return VmExpansionTile(
+          title: 'Retaining Path',
+          onExpanded: onExpanded,
+          children: retainingPath.value == null
+              ? [
+                  SizedBox.fromSize(
+                    size: const Size.fromHeight(
+                      2 * (defaultIconSizeBeforeScaling + denseSpacing),
+                    ),
+                    child: const CenteredCircularProgressIndicator(),
+                  )
+                ]
+              : _retainingPathList(
+                  context,
+                  retainingPath.value!,
+                ),
+        );
+      },
+    );
+  }
+}
+
+/// Returns a list of Widgets that will be the rows in the VmExpansionTile
+/// for RetainingPathWidget.
+List<Widget> _retainingPathList(
```

Comment:
This should be a private member of `RetainingPathWidget`

Reply:
Agreed. Thanks!